components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/SynapseAppDeployer.java

```diff
@@ -222,6 +256,453 @@ private void deployClassMediators(List<Artifact.Dependency> artifacts,
+     */
+    private void deploySynapseLibrary(List<Artifact.Dependency> artifacts,
+                                      AxisConfiguration axisConfig) throws DeploymentException {
+        for (Artifact.Dependency dependency : artifacts) {
+
+            Artifact artifact = dependency.getArtifact();
+            if (!validateArtifact(artifact)) {
+                continue;
+            }
+
+            if (SynapseAppDeployerConstants.SYNAPSE_LIBRARY_TYPE.equals(artifact.getType())) {
+
+                Deployer deployer = getSynapseLibraryDeployer(axisConfig);
+
+                if (deployer != null) {
+                    artifact.setRuntimeObjectName(artifact.getName());
+                    String fileName = artifact.getFiles().get(0).getName();
+                    String artifactPath = artifact.getExtractedPath() + File.separator + fileName;
+                    String artifactDir = getArtifactDirPath(axisConfig, SynapseAppDeployerConstants.SYNAPSE_LIBS);
+                    File artifactInRepo = new File(artifactDir + File.separator + fileName);
+                    if (artifactInRepo.exists()) {
+                        log.warn("Synapse Library " + fileName + " already found in " + artifactInRepo.getAbsolutePath() +
+                                ". Ignoring CAPP's artifact");
+                        artifact.setDeploymentStatus(AppDeployerConstants.DEPLOYMENT_STATUS_DEPLOYED);
+                    } else {
+                        try {
+                            deployer.deploy(new DeploymentFileData(new File(artifactPath), deployer));
+                            artifact.setDeploymentStatus(AppDeployerConstants.DEPLOYMENT_STATUS_DEPLOYED);
+                            try {
+                                String artifactName = getArtifactName(artifactPath, axisConfig);
+                                SynapseConfiguration configuration = getSynapseConfiguration(axisConfig);
+                                if (artifactName != null) {
+                                    if (configuration.getSynapseImports().get(artifactName) == null) {
+                                        String libName = artifactName.substring(artifactName.lastIndexOf("}") + 1);
+                                        String libraryPackage = artifactName.substring(1, artifactName.lastIndexOf("}"));
+                                        updateStatus(artifactName, libName, libraryPackage, ServiceBusConstants.ENABLED, axisConfig);
+                                    }
+                                }
+                            } catch (AxisFault axisFault) {
+                                log.error("Unable to update status for the synapse library : " + axisFault.getMessage());
+                            } catch (NullPointerException nullException) {
+                                log.error("Error while getting qualified name of the synapse library : " + nullException.getMessage());
```

Comment:
It should do a null check rather than catching NullPointerException (Runtime exception)